tests/engine/processes/workchains/test_restart.py

```diff
@@ -101,6 +105,10 @@ def test_unhandled_reset_after_handled(generate_work_chain, generate_calculation
     # Exit status 400 of the last calculation job will be handled and so should reset the flag
     process.ctx.children.append(generate_calculation_node(exit_status=400))
     result = process.inspect_process()
+
+    # Even though `handler_a` was followed by `handler_b`, we should retrieve the exit_code from `handler_a` because
+    # `handler_b` returned `None` which should not overwrite the last report.
     assert isinstance(result, ExitCode)
-    assert result.status == 0
+    assert result.status == 418
+    assert result.message == 'IMATEAPOT'
     assert process.ctx.unhandled_failure is False
```

Comment:
If we are testing the reset, and although it might be redundant because you already check that a calculation with an exit status != 0 sets off the `process.ctx.unhandled_failure`, shouldn't it be anyways better to check that the status actually change and was set back? (so perhaps `assert process.ctx.unhandled_failure is True` just before calling `result = process.inspect_process()`)

Reply:
I may misunderstand your point, but the `unhandled_failure` is only changed by the workchain in the `inspect_process` call. So adding the `assert process.ctx.unhandled_failure is True` check before the call will fail.

Reply:
No, I think you understood correctly. But then maybe the docstring of the test should say "Test `ctx.unhandled_failure` is set to `False` in `inspect_process` when a failed process is handled." or something like this? I mean, this is probably a nitpick, but using "reset" makes me think that at some point the `ctx.unhandled_failure` is set to `True`, which is not the case from what I am seeing in the code right?

Reply:
Ah I see now, you are right. There is a mistake. The idea of the test is that there is first a failed calculation, which sets `unhandled_failure=True` followed by another failed calculation, but that one returns a report, so it is not considered unhandled and so should reset the flag. The first calculation shouldn't be `0` then but some non-zero code. Will push a fix.

Reply:
Ah, ok, that makes more sense, haaha. This was why I was asking why you needed this previous calculation in the other PR. Good catch (Y)

Reply:
> Good catch (Y)

Right back at ya. Thanks for the review :+1: